### ISSUE TITLE
SIMD for faster calculations?

### ISSUE THREAD

**Issue by archuser555** [TheAlgorithms/Rust#357]
we can use SIMD to improve calculations and math speed

**Comment by er888kh** [comment]
The compiler is already really well at using SIMD, if the code is well written. On the other hand, having some SIMD code as a tutorial for instruction-level parallelism in Rust would be awesome!

We should keep in mind that the main focus of this repository is to give a pretty good implementation of some algos/DSs over having the best implementation by sacrificing readability or the educational aspect. If I were going to start reading OpenSSL's implementation of SHA-256, I would fail miserably. But it is arguably the most secure and performant implementation out there.

TLDR; We always love to have educational and well-written code, no matter how advanced it might be. But having _the best_ implementation out there is pretty much an anti-goal. Having said that, writing examples of SIMD hand-optimized code with good documentation can be both fun and educational!

**Comment by siriak** [comment]
Feel free to add separate implementations that use SIMD instructions. They'll reside next to usual algorithms and indicate with their name how they are different from conventional/SISD versions of the same algorithm.

**Comment by archuser555** [comment]
Ok Nice, thanks for the info's guys, i think i should close this, thanks again!
